### ISSUE TITLE
ERR_CONNECTION_REFUSED

### ISSUE THREAD

**Issue by dokbua** [linuxserver/docker-headphones#46]
[linuxserverurl]: https://linuxserver.io
[![linuxserver.io](https://raw.githubusercontent.com/linuxserver/docker-templates/master/linuxserver.io/img/linuxserver_medium.png)][linuxserverurl]

root@gw:~/docker/headphones# wget localhost:8181
--2019-09-24 23:09:27--  http://localhost:8181/
Resolving localhost (localhost)... ::1, 127.0.0.1
Connecting to localhost (localhost)|::1|:8181... connected.
HTTP request sent, awaiting response... Read error (Connection reset by peer) in headers.
Retrying.

--2019-09-24 23:09:28--  (try: 2)  http://localhost:8181/
Connecting to localhost (localhost)|::1|:8181... connected.
HTTP request sent, awaiting response... Read error (Connection reset by peer) in headers.
Retrying.

--2019-09-24 23:09:30--  (try: 3)  http://localhost:8181/
Connecting to localhost (localhost)|::1|:8181... connected.
HTTP request sent, awaiting response... Read error (Connection reset by peer) in headers.
Retrying.

Trying same in a browser cause it to flash and err with the error mentioned in the subject field.

	

Linux (Debian)

```
# cat docker-compose.yml
version: "2"
services:
  headphones:
    image: linuxserver/headphones
    container_name: headphones
    environment:
      - PUID=1000
      - PGID=1000
      - TZ=Europe/London
    volumes:
      - ./config:/config
      - ./downloads:/downloads
      - ./music:/music
    ports:
      - 8181:8181
    restart: unless-stopped
```

```
root@gw:~/docker/headphones# docker logs headphones
[s6-init] making user provided files available at /var/run/s6/etc...exited 0.
[s6-init] ensuring user provided files have correct perms...exited 0.
[fix-attrs.d] applying ownership & permissions fixes...
[fix-attrs.d] done.
[cont-init.d] executing container initialization scripts...
[cont-init.d] 10-adduser: executing...

-------------------------------------
          _         ()
         | |  ___   _    __
         | | / __| | |  /  \
         | | \__ \ | | | () |
         |_| |___/ |_|  \__/


Brought to you by linuxserver.io
We gratefully accept donations at:
https://www.linuxserver.io/donate/
-------------------------------------
GID/UID
-------------------------------------

User uid:    1000
User gid:    1000
-------------------------------------

[cont-init.d] 10-adduser: exited 0.
[cont-init.d] 30-config: executing...
[cont-init.d] 30-config: exited 0.
[cont-init.d] 99-custom-files: executing...
[custom-init] no custom files found exiting...
[cont-init.d] 99-custom-files: exited 0.
[cont-init.d] done.
[services.d] starting services
[services.d] done.
24-Sep-2019 22:07:56 - INFO :: MainThread : Checking to see if the database has all tables....
24-Sep-2019 22:07:56 - INFO :: MainThread : Retrieving latest version information from GitHub
24-Sep-2019 22:07:57 - INFO :: MainThread : Headphones is up to date
24-Sep-2019 22:07:57 - INFO :: MainThread : Starting Headphones web server on http://localhost:8181/
24-Sep-2019 22:07:57 - INFO :: MainThread : Scheduled background task: Search for Wanted
24-Sep-2019 22:07:57 - INFO :: MainThread : Scheduled background task: Download Scan
24-Sep-2019 22:07:57 - INFO :: MainThread : Scheduled background task: Library Scan
24-Sep-2019 22:07:57 - INFO :: MainThread : Scheduled background task: MusicBrainz Update
24-Sep-2019 22:07:57 - INFO :: MainThread : Scheduled background task: Check GitHub for updates
```

I'm not

**Comment by j0nnymoe** [comment]
Tried host ip?

**Comment by dokbua** [comment]
Yes, that's what I do on the browser. Same result.  Anyway, it's listening, I get a connecction refused, not a timeout.

**Comment by Eagle334** [comment]
I have the same issue. No way to have headphones work again. Try a purge (prune) and start from scratch. 

```
[cont-init.d] 10-adduser: exited 0.
[cont-init.d] 30-config: executing... 
[cont-init.d] 30-config: exited 0.
[cont-init.d] 99-custom-files: executing... 
[custom-init] no custom files found exiting...
[cont-init.d] 99-custom-files: exited 0.
[cont-init.d] done.
[services.d] starting services
[services.d] done.
19-Oct-2019 18:25:32 - INFO :: MainThread : Checking to see if the database has all tables....
19-Oct-2019 18:25:32 - INFO :: MainThread : Retrieving latest version information from GitHub
19-Oct-2019 18:25:37 - ERROR :: MainThread : Unable to connect to remote host. Check if the remote host is up and running.
19-Oct-2019 18:25:37 - WARNING :: MainThread : Could not get the latest version from GitHub. Are you running a local development version?
19-Oct-2019 18:25:37 - INFO :: MainThread : Starting Headphones web server on http://0.0.0.0:8181/
19-Oct-2019 18:25:38 - INFO :: MainThread : Scheduled background task: Search for Wanted
19-Oct-2019 18:25:38 - INFO :: MainThread : Scheduled background task: Download Scan
19-Oct-2019 18:25:38 - INFO :: MainThread : Scheduled background task: Library Scan
19-Oct-2019 18:25:38 - INFO :: MainThread : Scheduled background task: MusicBrainz Update
19-Oct-2019 18:25:38 - INFO :: MainThread : Scheduled background task: Check GitHub for updates
```

```
wget http://localhost:8181/
--2019-10-19 18:39:52--  http://localhost:8181/
Resolving localhost (localhost)... 127.0.0.1, ::1
Connecting to localhost (localhost)|127.0.0.1|:8181... connected.
HTTP request sent, awaiting response... Read error (Connection reset by peer) in headers.
Retrying.
```

**Comment by D-e-n-t** [comment]
Anybody know how to fix this?

**Comment by tanc** [comment]
(edited) I have the exact same issue, output is as Eagle334 above. Container is running fine on Linux server but not accessible on 8181. Set up locally on Linux (Arch) and it works, the web interface is available on 8181

**Comment by tanc** [comment]
I was able to solve my issue. I had to specify the ports explicitly. I'm using ansible to set up the container but however you do it, when you create the container you must specify the ports `8181:8181`

**Comment by skupjoe** [comment]
I was getting timeouts/resets, but it was because of the way I was networking and routing 8181 traffic on my docker host.

Specifically, in my case, I added a VPN network for my headphones container and mangling marked packets out the VPN vif on my host, but I needed to add some iptables RETURN statements to bypass this for local 8181 traffic:

```
iptables -t mangle -I PREROUTING 1 -p tcp --dport 8181 -j RETURN
iptables -t mangle -I PREROUTING 1 -p tcp --sport 8181 -j RETURN
```

Maybe this will help somebody out there! More than likely whatever is causing these issues is network-related and different for each person.
